scripts/create_burndown.py

```diff
@@ -17,211 +17,402 @@
+      self.open_tasks.append(day["tasks"]["open"])
+      self.total_tasks.append(day["tasks"]["total"])
+      self.max_story_points = max(self.max_story_points, day["story_points"]["total"])
+    
+      # TODO: refactor to not to draw 0 extra Story Points Line if one task
+      # in extra card been done. Example http://imagebin.suse.de/1785/img
+      if "story_points_extra" in day or "tasks_extra" in day:
+        self.extra_days.append(current_day)
+        tasks = 0
+        if day.has_key("tasks_extra"):
+          tasks = -day["tasks_extra"]["done"]
+        self.bonus_tasks_done.append(tasks)
+        points = 0
+        if day.has_key("story_points_extra"):
+          points = -day["story_points_extra"]["done"]
+        self.bonus_story_points_done.append(points)
+    
+      if day.has_key("fast_lane"):
+        self.x_fast_lane.append(current_day)
+        self.y_fast_lane.append(day["fast_lane"]["open"])
+        self.total_fast_lane.append(day["fast_lane"]["total"])
+    
+      current_day += 1
+  
+    # If there are already done tasks at the beginning of a sprint
+    if burndown["days"][0].has_key("tasks_extra"):
+       self.bonus_tasks_day_one = burndown["days"][0]["tasks_extra"]["done"]
+    
+    # Add a day at the beginning of the extra days, so the curve starts at zero
+    if len(self.extra_days) > 0 and not self.bonus_tasks_day_one:
+      self.extra_days = [self.extra_days[0] - 1] + self.extra_days
+      self.bonus_story_points_done = [0] + self.bonus_story_points_done
+      self.bonus_tasks_done = [0] + self.bonus_tasks_done
+      self.extra_day = 1
+
+    return
+
+
+class PlotConfig:
+  "Set all parameters needed to print burndown charts"
+
+  def __init__ (self, data):
+    figure_width = 11
+    figure_height = 6
+    
+    title_fontsize = 'large'
+    label_x_axis = 'Days'
+    
+    color_diagonal = 'grey'
+    color_waterline = 'blue'
+    color_weekend = 'grey'
+    
+    linestyle_waterline = ':'
+    linestyle_weekend = ':'
+
+    self.setXKCD()
+    self.setYAxisLimits(data.max_story_points, data.bonus_tasks_done, data.bonus_story_points_done)
+    self.createFigure(figure_width, figure_height)
+    self.setTitle(data.sprint_number, title_fontsize)
+    self.setXAxisLabel(label_x_axis)
+    self.plotDiagonal(1, data.total_days, data.max_story_points, 0, color_diagonal)
+    self.plotWaterLine(0, data.total_days+1, 0, 0, color_waterline, linestyle_waterline)
+    self.plotWeekendLines(data.weekend_lines, self.ymin+1, self.ymax-1, color_weekend, linestyle_weekend)
+    self.setScaleFactor(data.total_tasks[0], data.max_story_points)
+    self.setXAxisLimits(0, data.total_days+1)
+    self.setXAxisTicks(data.days)
+
+  # Plot the chart in XKCD style
+  def setXKCD(self):
+    plt.xkcd()
+    return
+
+  # Calculate minimum and maximum values for the y axis
+  def setYAxisLimits(self, max_story_points, bonus_tasks_done, bonus_story_points_done):
```

Comment:
Maybe better call the function `calculateYRange` and let it return the calculated values.